src/core/or/circuitbuild.c

```diff
@@ -1244,13 +1244,15 @@ circuit_extend(cell_t *cell, circuit_t *circ)
     return -1;
   }
 
-  if (!ec.orport_ipv4.port || tor_addr_is_null(&ec.orport_ipv4.addr)) {
+  if ((!ec.orport_ipv4.port || tor_addr_is_null(&ec.orport_ipv4.addr)) &&
```

Comment:
This change is premature, relays do not extend over IPv6, so they require a non-null IPv4 extend address.

Please make a separate block which rejects any IPv6 extend.